### PR TITLE
Update nomicon on drop flags

### LINKED ISSUE #38481: Outdated info in nomicon on drop flags
> As of Rust 1.0, the drop flags are actually not-so-secretly stashed in a hidden field of any type that implements Drop. Rust sets the drop flag by overwriting the entire value with a particular bit pattern. This is pretty obviously Not The Fastest and causes a bunch of trouble with optimizing code. It's legacy from a time when you could do much more complex conditional initialization.

> As such work is currently under way to move the flags out onto the stack frame where they more reasonably belong. Unfortunately, this work will take some time as it requires fairly substantial changes to the compiler.

[https://github.com/rust-lang/rust/blob/master/src/doc/nomicon/drop-flags.md](https://github.com/rust-lang/rust/blob/master/src/doc/nomicon/drop-flags.md)

The work described landed in #33622, so this is now out of date.

### PR DESCRIPTION
**Pull request by sanxiyn**
Fix #38481.

r? @steveklabnik

### CODE REVIEW

**Comment by steveklabnik** [comment]
Thank you! However, in documentation, we generally don't talk about versions of Rust like this. Could you instead cut everything but the actual explanation of how it's done today? Thanks!

**Comment by liigo** [comment]
Version is useful information here.

**Comment by liigo** [comment]
You mentioned version is stable version, right?

**Comment by steveklabnik** [comment]
Docs are always written in the present tense; we distribute them with each version of Rust, so if you wanted to see how something works on a particular version, you'd download those docs. This is of course, modulo bugs like this. Regardless, including "since version x.y.z" isn't how we write any docs, so this should be in line with the rest of Rust's documentation.

**Comment by petrochenkov** [comment]
> "since version x.y.z" isn't how we write any docs

It's never late to start!
A small piece of history looks useful and fitting in this context, why throw it away in favor of some formalities.

**Comment by steveklabnik** [comment]
Because we believe that it's a better way to write documentation.

Anyway, this isn't the right place to argue about it; a policy change here would need a docs RFC, so if you'd like this to land before then, you gotta make it fit with the current way we write docs, sorry.

**Comment by steveklabnik** [comment]
ping @sanxiyn ! Can you make the change I requested, please?

File: src/doc/nomicon/drop-flags.md [hunk 1]
```diff
@@ -79,15 +79,14 @@ if condition {
...
-where they more reasonably belong. Unfortunately, this work will take some time
-as it requires fairly substantial changes to the compiler.
+In Rust 1.0, the drop flags were actually not-so-secretly stashed in a hidden
+field of any type that implements Drop. Rust set the drop flag by overwriting
+the entire value with a particular bit pattern. This was pretty obviously Not
+The Fastest and caused a bunch of trouble with optimizing code. It was legacy
+from a time when you could do much more complex conditional initialization.
+
+Since Rust 1.13, the flags has been moved to the stack frame where they more
+reasonably belong, and types implementing Drop no more take up any extra space.
```
Comment (pnkfelix) [inline src/doc/nomicon/drop-flags.md]: nit: "no longer take up any ...", not "no more take up any ..."

**Comment by sanxiyn** [comment]
I am not sure how best to reword this, maybe someone else can do so. Thanks for review!

**Pull request closed without merging**